d1985r0.md

```diff
@@ -394,6 +394,94 @@ Z<int> z1; // ok, decltype(z1)::value = int
 Z<1> z; // error, alias<1>::value is not a type
 ```
 
+## Overloading of class templates
+
+Universal template arguments enable what appears like overloading of
+class templates by implementing each overload as a specialization of a
+unimplemented primary template. As this is not real overloading there will be some
+differences, especially when it comes to CTAD.
+
+```cpp
+template<template auto...> class my_container;
+
+template<typename T> class my_container<T> {
+public:
+    my_container(T* data, size_t count);
+    // A basic implementation
+};
+
+template<typename T, typename A> class my_container<T, A> {
+    my_container(T* data, size_t count);
+    my_container(T* data, size_t count, const A& alloc);
+    // An implementation using an allocator A
+};
+
+template<typename T, size_t SZ> class my_container<T, SZ> {
+    my_container(T* data, size_t count);
+    // An implementation with an internal storage of SZ bytes
+};
+
+template<typename T> my_container(T*, size_t) -> my_container<T>;
+template<typename T, typename A> my_container(T*, size_t, const A&) -> my_container<T, A>;
+
+```
+
+As the primary template is not defined there is only a default
+constructor implicit deduction guide. The explicit deduction guides
+select the first specialization unless an allocator object is given
+in which case the second specialization is selected. To select the
+third specialization the template arguments must be explicitly given.
+
+
+## Overloading of variable templates
```

Comment:
"Impacts on the specialization of variable templates"

Reply:
Nice way of putting it...